src/document-submit.js

```diff
@@ -55,18 +55,19 @@ export function onDocumentFormSubmit_(e) {
   const inputs = form.elements;
   for (let i = 0; i < inputs.length; i++) {
     user().assert(!inputs[i].name ||
-        inputs[i].name.indexOf(SOURCE_ORIGIN_PARAM) == -1,
+        inputs[i].name != SOURCE_ORIGIN_PARAM,
         'Illegal input name, %s found: %s', SOURCE_ORIGIN_PARAM, inputs[i]);
   }
 
   const win = form.ownerDocument.defaultView;
   const action = form.getAttribute('action');
-  user().assert(action, 'form action attribute is required: %s', form);
-  assertHttpsUrl(action, form, 'action');
-  user().assert(!startsWith(action, urls.cdn),
+  form.__AMP_INIT_ACTION__ = form.__AMP_INIT_ACTION__ || action;
```

Comment:
Not a big fun of using `__AMP_INIT_ACTION_` through the code. I'd rather it was read/written once and then operate with a local var.